cluster-api/pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -107,7 +107,24 @@ type ProviderConfig struct {
 // ProviderConfigSource represents a source for the provider-specific
 // node configuration.
 type ProviderConfigSource struct {
-	// TODO(roberthbailey): Fill these in later
+
+	// No more than one of the following may be specified.
+
+	// The machine class from which the provider config should be sourced.
+	// +optional
+	MachineClass *MachineClassRef `json:machineClass,omitempty`
+}
+
+type MachineClassRef struct {
+	// The name of the MachineClass.
+	Name string `json:name`
+
```

Comment:
What about namespace?

Reply:
Storage classes aren't namespaced, and I used that same pattern here. But it's worth discussing whether they should be part of a namespace.